pyscriptjs/src/plugins/pyterminal.ts

```diff
@@ -68,18 +79,48 @@ export class PyTerminalPlugin extends Plugin {
         //
         //   4. (in the future we might want to add an option to start the
         //      capture earlier, but I don't think it's important now).
-        const PyTerminal = make_PyTerminal(this.app);
+        const PyTerminal = _interpreter.config.xterm ? make_PyTerminal_xterm(this.app) : make_PyTerminal_pre(this.app);
         customElements.define('py-terminal', PyTerminal);
```

Comment:
That's my intention, yeah - exactly as you say, since we (currently) only have a single global place to determine whether a py-terminal is xterm or not, I reckon all terminals across a single page should be one or the other.

Relatedly, all py-terminals on a page currently get all writes to _stdout_ from our (one) interpreter. I would think if you're writing code that's meant to go to an xterm, you need _all_ of the terminals on the page to be an xterm.

Probably something good for us to revisit if/when we have multiple interpreters on the page though.

Reply:
Definitively so much else to change if that's the case but the thing here is: we can't have different behavior registered as same custom element (by name) ... which makes me wonder if we could instead provide a `py-xterm` instead of dancing with the registry to make `py-terminal` somehow *ambiguous* (where by ambiguous, I mean it can be something else, like an *xterm* instead) ... if you're confident this is not an issue though, I'm OK, as long as we understand this dance can't change in the future without breaking the registry if we try to host multiple interpreters in the same page. Resolving for now as this seems to be well understood.